tests/test_daemon_query.py

```
@@ -0,0 +1,284 @@
+                except SystemExit:
+                    pass
+
+            # Verify complex payload was parsed correctly
+            mock_query_func.assert_called_once()
+            called_with_project, called_with_command = mock_query_func.call_args[0]
+            assert called_with_project == project_path
+            assert called_with_command == json.loads(complex_json)
+            assert called_with_command["filters"]["language"] == "python"
+            assert called_with_command["limit"] == 10
+
+    def test_query_json_with_special_characters(self, capsys):
```

Ruff (0.14.11)

Unused method argument: `capsys`

(ARG002)